scipy/conftest.py

```diff
@@ -151,6 +151,10 @@ def check_fpu_mode(request):
 
 array_api_compatible = pytest.mark.parametrize("xp", xp_available_backends.values())
 
+skip_array_api_invalid_arg = pytest.mark.skipif(SCIPY_ARRAY_API,
+    reason = ('Test involves masked arrays, object arrays, or other '
+              'types that are not valid input when SCIPY_ARRAY_API=1'))
```

Comment:
not _quite_ true, given that this is the definition of `SCIPY_ARRAY_API`:

```
SCIPY_ARRAY_API: str | bool = os.environ.get("SCIPY_ARRAY_API", False)
```

I would just say "when `SCIPY_ARRAY_API` is set".

Reply:
Ok, but is it intended for `SCIPY_ARRAY_API = 0` to have the same behavior as `1`? I've noted elsewhere that `SCIPY_ARRAY_API = 0` results in errors, but I've assumed this is a mistake.

Reply:
In that case, let's not change this documentation. Setting it to `0` doesn't and should not do the same as setting it to `1`, and I don't know that we'd want to advertise that it can be any truthy value if it's supposed to be a boolean.
I can file an issue about the error when it is set to `0`.

Reply:
I misunderstood originally - if the distinction was that "True" is also a valid input, I don't think we would need to change this. However, `SCIPY_ARRAY_API` can be JSON, which is different than `True`, but presumably it also causes xp-incompatible input to fail. So I changed the wording to "types that are not valid input when `SCIPY_ARRAY_API` is _used_", which leaves a little room for interpretation compared to "set".

Reply:
Yes, that sounds better than what I had in mind, thanks.